### PR TITLE
Release audio hold when media service reset

### PR DESCRIPTION
**Pull request by ziyiz-amzn**
### Issue #, if available:
Switching audio devices makes audio session to stop. 
https://github.com/aws/amazon-chime-sdk-ios/issues/322

### Description of changes:
Add an observer that handle AVAudioSession.mediaServicesWereResetNotification, in which it calls audioClientController.endOnHold() to reclaim the audio device.

### Testing done:
1. Start meeting
2. Connect AirPods
3. Change to phone speaker
4. Switch back to AirPods
5. Disconnect AirPods by turn off bluetooth
6. Audio session keeps live

#### Manual test cases (add more as needed):

- [x] Join meeting without CallKit
- [ ] Join meeting with CallKit as Incoming
- [ ] Join meeting with CallKit as Outgoing
- [x] Leave meeting
- [ ] Rejoin meeting
- [ ] See metrics
- [ ] See attendee presence
- [x] Send audio
- [x] Receive audio
- [x] Mute self
- [x] Unmute self
- [ ] See local mute indicator when muted
- [ ] See remote mute indicator when other is muted
- [ ] Enable and disable local video
- [ ] Enable and disable remote video from remote side
- [ ] Send local video
- [ ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Receive remote screen share

#### Integration validation (required before release)

- [ ] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

### Screenshots, if available:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonChimeSDK/AmazonChimeSDK/audiovideo/DefaultAudioVideoController.swift [hunk 1]
```diff
@@ -94,4 +103,14 @@ import Foundation
     public func stopRemoteVideo() {
         videoClientController.stopRemoteVideo()
     }
+
+    @objc private func handleMediaServiceReset(notification: Notification) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(1)) {
```
Comment (zhinang-amazon) [inline AmazonChimeSDK/AmazonChimeSDK/audiovideo/DefaultAudioVideoController.swift:108]: why the 1sec delay?

Comment (ziyiz-amzn) [reply AmazonChimeSDK/AmazonChimeSDK/audiovideo/DefaultAudioVideoController.swift:108]: Discussed with Nick, wanted to release audio when reset is completed. Tested with no delay, it works the same though.

Comment (ziyiz-amzn) [reply AmazonChimeSDK/AmazonChimeSDK/audiovideo/DefaultAudioVideoController.swift:108]: I tested a few more times without delay, no unexpected behaviors. Will remove it.

**Review by zhinang-amazon — DISMISSED** [review]

File: AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift [hunk 2]
```diff
@@ -161,4 +161,8 @@ extension DefaultAudioClientController: AudioClientController {
         }
```
Comment (hokyungh) [inline AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift:161]: why not add `NotificationCenter.default` here?

**Review by hokyungh — APPROVED** [review]

File: AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift (hunk 2 above)
Comment (ziyiz-amzn) [reply AmazonChimeSDK/AmazonChimeSDK/internal/audio/DefaultAudioClientController.swift:169]: Resolved.

**Review by zhinang-amazon — APPROVED** [review]

**Review by hokyungh — APPROVED** [review]

**Pull request merged**